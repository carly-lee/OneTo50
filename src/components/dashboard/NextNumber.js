import React, { PureComponent } from 'react';
import { Text, View, StyleSheet } from 'react-native';

//@flow
export default class NextNumber extends PureComponent{
  render(){
    return(
      <View style={ styles.container }>
        <Text>1</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 0.4,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
});