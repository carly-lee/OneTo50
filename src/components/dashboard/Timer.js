import React, { PureComponent } from 'react';
import { Text, View, StyleSheet } from 'react-native';

//@flow
type Props = {
  reportFinalTime: ( time:number )=> void,
  nextNumber: number,
}

export default class Timer extends PureComponent{
  props:Props;
  timer: ?number = null;
  state: {
    currentTime: number,
  };

  state = {
    currentTime: 0,
  }

  componentWillReceiveProps( nextProps ){
    if( nextProps.nextNumber > 50 ){
      clearInterval( this.timer );
      this.timer = null;
      this.props.reportFinalTime( this.state.currentTime );
    }
  }

  componentDidMount(){
    this.timer = setInterval(()=>{
      this.setState(( prevState )=>{
        console.debug( 'timer' );
        const currentTime:number = Number( prevState.currentTime ) + 0.01;
        return { currentTime: currentTime.toFixed( 2 ) };
      });
    }, 10 );
  }

  render(){
    return(
      <View style={ styles.container }>
        <Text>{this.state.currentTime}</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 0.6,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'pink',
  },
});