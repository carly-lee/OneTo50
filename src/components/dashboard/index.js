export { default as Dashboard } from './Dashboard';
export { default as Timer } from './Timer';
export { default as NextNumber } from './NextNumber';