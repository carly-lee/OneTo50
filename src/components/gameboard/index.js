export { default as GameBoard } from './GameBoard';
export { default as NumberButton } from './NumberButton';