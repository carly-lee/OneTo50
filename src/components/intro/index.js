// @flow
// $FlowIgnore
import { Text, View, StyleSheet, Button } from 'react-native';
import React, { PureComponent } from 'react';

type Props = {
  onPress: () => void,
};

export default class Intro extends PureComponent{
  props: Props;

  _onPressStart = ():void =>{
    this.props.onPress();
  }

  render(){
    return(
      <View style={ styles.container }>
        <Text style={ styles.title }>
          1 to 50
        </Text>
        <Button
          onPress={ this._onPressStart }
          title="Let's Start!"
          color="#841584"
          accessibilityLabel="Let's Start!"
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  title: {
    fontSize: 48,
    fontWeight: 'bold',
    textAlign: 'center',
  },
});