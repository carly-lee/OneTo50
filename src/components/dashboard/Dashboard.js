import React, { PureComponent } from 'react';
import { View, StyleSheet } from 'react-native';

import { Timer, NextNumber } from 'components/dashboard';

//@flow
export default class Dashboard extends PureComponent{
  render(){
    return(
      <View style={ styles.container }>
        <Timer />
        <NextNumber />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 0.2,
    flexDirection: 'row',
    justifyContent: 'space-between',
    backgroundColor: 'skyblue',
  },
});