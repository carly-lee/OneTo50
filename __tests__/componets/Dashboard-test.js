import 'react-native';
import React from 'react';
import { Dashboard } from 'components/dashboard';
import renderer from 'react-test-renderer';

describe( 'Dashboard', ()=>{
  it( 'renders correctly',()=>{
    const tree = renderer.create( <Dashboard /> );
    expect( tree ).toMatchSnapshot();
  });
});