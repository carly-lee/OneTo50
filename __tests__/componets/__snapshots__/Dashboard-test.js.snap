exports[`Dashboard renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "skyblue",
      "flex": 0.2,
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "pink",
        "flex": 0.6,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail">
      0
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 0.4,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail">
      1
    </Text>
  </View>
</View>
`;
